src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -121,13 +124,16 @@ public BitArray(bool[] values)
                 throw new ArgumentNullException(nameof(values));
             }
 
-            m_array = new int[GetArrayLength(values.Length, BitsPerInt32)];
+            m_array = new int[GetIntegerArrayLengthFromBitLength(values.Length)];
             m_length = values.Length;
 
             for (int i = 0; i < values.Length; i++)
             {
                 if (values[i])
-                    m_array[i / 32] |= (1 << (i % 32));
+                {
+                    int quotient = Div32Rem(i, out int remainder);
+                    m_array[quotient] |= 1 << remainder;
```

Comment:
Does this help even though the dividend for both operations was a constant power of 2?

Reply:
Will measure & verify. Looking at the results from [sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgBgARYCMA3ALABQWAzAQEx4DCeA3hXuwTVgCx4CyhABQBLAHYwA2gF08YAPoQECCAE8keMTA0BKFmw4H5i5SonC8AejxVaMjAF48gwngA8rp+YCk12tu1k5AYAvvrsYZwEvHy0IuLSsgpKquqaOnpBBhxpAI4ArgD2MMJw4niO5gB8lXgArIFZ2WUIcJBiACaI5Rp4ALRO+UUlZe51ARGGSSYSg8WlMHaOzm4egi1top0I/g0coZnhB5E8/FRxkjJGyWoaZcK6rEeTxqpmlniEAJyLTi6jIngfF8dhF9gYItQovxuOcElcTKk7g8Jk0tLNhloKu8vrtGml1hAOl0sf1BOj5ngAFQfT7jJ4ceGvcniH7Lf4EonbOkhCjBIA=) though, it is only powers of 2 that actually benefit from such an optimzation (in terms of number of instructions), since we can leverage bit-wise operations. Non-powers have 2 have the same (or larger) disassembly size.

Reply:
``` ini

BenchmarkDotNet=v0.11.2, OS=macOS High Sierra 10.13.3 (17D47) [Darwin 17.4.0]
Intel Core i7-4870HQ CPU 2.50GHz (Haswell), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=2.1.300-preview2-008062
  [Host]     : .NET Core ? (CoreCLR 4.6.27019.06, CoreFX 4.6.27019.05), 64bit RyuJIT
  DefaultJob : .NET Core 2.1.0-preview2-26130-06 (CoreCLR 4.6.26130.05, CoreFX 4.6.26130.01), 64bit RyuJIT


```
|    Method |     Mean |     Error |    StdDev | Ratio |
|---------- |---------:|----------:|----------:|------:|
| DivAndMod | 19.96 us | 0.0407 us | 0.0340 us |  1.00 |
|  Div32Rem | 16.27 us | 0.0267 us | 0.0237 us |  0.81 |

```C#
public class BitArrayPerf
{
    private const int N = 10_000;
    private readonly int[] data;

    public BitArrayPerf()
    {
        data = new int[N];
        var rand = new Random(42);
        for (int i = 0; i < N; i++)
        {
            data[i] = rand.Next();
        }
    }

    [Benchmark(Baseline = true)]
    public int DivAndMod()
    {
        for (int i = 0; i < N; i++)
        {
            data[i / 32] |= (1 << (i % 32));
        }
        return data[N - 1];
    }

    [Benchmark]
    public int Div32Rem()
    {
        for (int i = 0; i < N; i++)
        {
            int quotient = Div32Rem(i, out int remainder);
            data[quotient] |= 1 << remainder;
        }
        return data[N - 1];
    }

    private static int Div32Rem(int number, out int remainder)
    {
        int quotient = number >> 5;   // Divide by 32.
        remainder = number - (quotient << 5);   // Multiply by 32.
        return quotient;
    }
}
```

Reply:
And here is the disassembly: https://www.diffchecker.com/Yirmy9Pk

30 vs 36 instructions